x-pack/plugins/ml/public/application/datavisualizer/data_drift/index_patterns_picker.tsx

```diff
@@ -0,0 +1,219 @@
+              onChoose={onObjectSelection}
+              showFilter
+              noItemsMessage={i18n.translate(
+                'xpack.ml.newJob.wizard.searchSelection.notFoundLabel',
+                {
+                  defaultMessage: 'No matching data views or saved searches found.',
+                }
+              )}
+              savedObjectMetaData={[
+                {
+                  type: 'search',
+                  getIconForSavedObject: () => 'search',
+                  name: i18n.translate(
+                    'xpack.ml.newJob.wizard.searchSelection.savedObjectType.search',
+                    {
+                      defaultMessage: 'Saved search',
+                    }
+                  ),
+                },
+                {
+                  type: 'index-pattern',
+                  getIconForSavedObject: () => 'indexPatternApp',
+                  name: i18n.translate(
+                    'xpack.ml.newJob.wizard.searchSelection.savedObjectType.dataView',
+                    {
+                      defaultMessage: 'Data view',
+                    }
+                  ),
+                },
+              ]}
+              fixedPageSize={20}
+              services={{
+                contentClient: contentManagement.client,
+                uiSettings,
+              }}
+            >
+              <EuiButton
+                size="m"
+                fill
+                iconType="plusInCircleFilled"
+                onClick={() => navigateToPath(createPath(ML_PAGES.DATA_DRIFT_CUSTOM))}
+                disabled={!canEditDataView}
+              >
+                <FormattedMessage
+                  id="xpack.ml.dataDrift.createDataViewButton"
+                  defaultMessage="Create a data view"
+                />
+              </EuiButton>
+            </SavedObjectFinder>
+          </EuiPanel>
+        </>
+      </EuiPageBody>
+    </div>
+  );
+};
+
+export const DataDriftIndexPatternsPicker: FC = () => {
+  const { sourceIp, destIp } = parse(location.search, {
+    sort: false,
+  }) as { sourceIp: string; destIp: string };
+
+  const [dataViewEditorServices, setDataViewEditorServices] = useState<
+    | {
+        referenceDataViewEditorService: DataViewEditorServiceSpec;
+        productionDataViewEditorService: DataViewEditorServiceSpec;
+      }
+    | undefined
+  >();
+
+  const {
+    services: {
+      dataViewEditor,
+      http,
+      data: { dataViews },
+    },
+  } = useMlKibana();
+  const { dataViewEditorServiceFactory } = dataViewEditor;
+
+  const initialProductionIndexPattern = useMemo(
+    () => (destIp ? destIp.replaceAll(`'`, '') : ''),
+    [destIp]
+  );
+  const initialReferenceIndexPattern = useMemo(
+    () => (sourceIp ? sourceIp.replaceAll(`'`, '') : ''),
+    [sourceIp]
+  );
+
+  useEffect(() => {
+    let unmounted = false;
+    const getDataViewEditorService = async () => {
+      if (!unmounted && http && dataViews && dataViewEditorServiceFactory) {
```

Comment:
you check for unmounted before the async action is settled, so it's not gonna work

Reply:
[Updated here](https://github.com/elastic/kibana/pull/162853/commits/2462e8f9eedfd1fdc171f6826367973741aa2b76)